### PR TITLE
Remove ONNXModelNormalizer

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

As stated in the title

### Reason for changes

Improve user API

### Related tickets

No

### Tests

No

### CODE REVIEW

File: nncf/experimental/onnx/graph/nncf_graph_builder.py [hunk 1]
```diff
@@ -36,6 +37,33 @@ class GraphConverter:
 
     DEFAULT_TENSOR_SHAPE = [1]
 
+    @staticmethod
+    def _preprocess_model(model: onnx.ModelProto) -> onnx.ModelProto:
+        model = GraphConverter._replace_empty_node_name(model)
+        return model
```
Comment (andrey-churkin) [inline nncf/experimental/onnx/graph/nncf_graph_builder.py:43]: This method is not needed.

File: nncf/experimental/onnx/graph/nncf_graph_builder.py [hunk 2]
```diff
@@ -36,6 +37,33 @@ class GraphConverter:
...
+
+    @staticmethod
+    def _replace_empty_node_name(model: onnx.ModelProto) -> onnx.ModelProto:
+        """
+        Sets a unique name to every node in 'model' with empty name field.
+        NNCFGraph expects every node to have a unique name.
+        :param model: ONNX model.
```
Comment (andrey-churkin) [inline nncf/experimental/onnx/graph/nncf_graph_builder.py:50]:
```suggestion
        NNCFGraph expects every node to have a unique name.

        :param model: ONNX model.
```

File: nncf/experimental/onnx/graph/nncf_graph_builder.py [hunk 3]
```diff
@@ -36,6 +37,33 @@ class GraphConverter:
...
+        :return: ONNX model with filled nodes.
+        """
+        for i, node in enumerate(model.graph.node):
+            if node.name == '':
+                node.name = node.op_type + '_nncf_' + str(i)
```
Comment (andrey-churkin) [inline nncf/experimental/onnx/graph/nncf_graph_builder.py:55]: I think it is better don't add 'nncf' here.
```python
node.name = f'{node.op_type}_{str(i)}'
```
is enough.

File: nncf/experimental/quantization/compression_builder.py [hunk 4]
```diff
@@ -80,26 +74,25 @@ def apply(self, model: TModel, dataset: Dataset) -> TModel:
         5) Collect all statistics.
         6) Apply algorithms.
         """
-
         if not self.algorithms:
             nncf_logger.info('There are no algorithms added. The original model will be returned.')
             return model
 
-        backend = get_backend(model)
+        _model = deepcopy(model)
```
Comment (andrey-churkin) [inline nncf/experimental/quantization/compression_builder.py:81]: Why is it necessary?

File: tests/onnx/utils.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+TARGET_OPSET_VERSION = 13
+TARGET_IR_VERSION = 7
+
+
+def convert_opset_version(model: onnx.ModelProto, opset_version: int = TARGET_OPSET_VERSION) -> onnx.ModelProto:
+    """
+    Tries to convert 'model' Opset Version to 'opset_version'.
+    If the 'model' can not be converted returns the original 'model'.
+    :param model: ONNX model to convert.
```
Comment (andrey-churkin) [inline tests/onnx/utils.py:13]:
```suggestion
    If the 'model' can not be converted returns the original 'model'.

    :param model: ONNX model to convert.
```

**Review by andrey-churkin — CHANGES_REQUESTED** [review]

File: nncf/experimental/quantization/compression_builder.py (hunk 4 above)
Comment (kshpv) [reply nncf/experimental/quantization/compression_builder.py:81]: The user doesn't expect that the original model that he passed will be changed, I suppose. 
We should work on a copy of that.

File: nncf/experimental/onnx/graph/nncf_graph_builder.py (hunk 1 above)
Comment (kshpv) [reply nncf/experimental/onnx/graph/nncf_graph_builder.py:43]: Done

File: nncf/experimental/onnx/graph/nncf_graph_builder.py (hunk 2 above)
Comment (kshpv) [reply nncf/experimental/onnx/graph/nncf_graph_builder.py]: Done

File: tests/onnx/utils.py (hunk 5 above)
Comment (kshpv) [reply tests/onnx/utils.py]: Done

**Comment by andrey-churkin** [comment]
LGTM, Could you please fix the ONNX precommit?

**Review by KodiaqQ — APPROVED** [review]

File: tests/onnx/utils.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
+"""
```
[the remark this answers is not in the document]
Comment (kshpv) [reply tests/onnx/utils.py]: Done

**Review by vshampor — APPROVED** [review]

File: nncf/quantization/algorithms/post_training/algorithm.py [hunk 7]
```diff
@@ -136,28 +138,19 @@ def _create_statistics_aggregator(self,
...
-
-        return None
-
     def _apply(self,
                model: TModel,
                statistic_points: Optional[StatisticPointsContainer] = None,
                dataset: Optional[Dataset] = None) -> TModel:
 
-        modified_model = model
+        modified_model = deepcopy(model)
```
Comment (alexsu52) [inline nncf/quantization/algorithms/post_training/algorithm.py:146]: Could you briefly explain why you are doing a deep copy?

**Review by alexsu52 — APPROVED** [review]
LGTM

File: nncf/quantization/algorithms/post_training/algorithm.py (hunk 7 above)
Comment (andrey-churkin) [reply nncf/quantization/algorithms/post_training/algorithm.py:146]: I asked the same question.
Answer was:
> The user doesn't expect that the original model that he passed will be changed, I suppose.
We should work on a copy of that.

**Pull request merged**